### ISSUE TITLE
`no-undef-properties` False positive where `defineModel` is used

### ISSUE THREAD

**Issue by Ky6uk** [vuejs/eslint-plugin-vue#2363]
**Tell us about your environment**

- **ESLint version:** 8.56.0
- **eslint-plugin-vue version:** 9.20.0
- **Node version:** 18.18.0

**Please show your full configuration:**

```js
module.exports = {
  extends: ['plugin:vue/base'],

  rules: {
    'vue/no-undef-properties': 'error',
  },
};
```

**What did you do?**	

```vue
<template>
  <div id="app">Woof: {{ woof }}</div>
</template>

<script setup>
const model = defineModel();

const woof = computed(() => model.value);
</script>
```

**What did you expect to happen?**

`vue/no-undef-properties` should not throw error.

**What actually happened?**

```
/home/projects/stackblitz-starters-u1b2em/test.vue
  8:35  error  'value' is not defined  vue/no-undef-properties

✖ 1 problem (1 error, 0 warnings)
```

**Repository to reproduce this issue**

https://stackblitz.com/edit/stackblitz-starters-u1b2em
